pallets/connectors-gateway/connectors-gateway-axelar-precompile/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use fp_evm::PrecompileHandle;
+use frame_support::dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo};
+use pallet_connectors_gatway::GatwayOrigin;
+use precompile_utils::prelude::*;
+use sp_core::{ConstU32, Get, H160, U256};
+
+/// Precompile implementing IAxelarForecallable.
+/// MUST be used as the receiver of calls over the axelar bridge
+pub struct AxelarForecallable<Runtime, Gateway, MaxPayload>(
+	core::marker::PhantomData<(Runtime, Gateway, MaxPayload)>,
+);
+
+#[precompile_utils::precompile]
```

Comment:
Do we need any other "EVM glue" for this to work or are these macro invocations enough to "register" this contract in the EVM?

Reply:
They are enough to `impl Precompile` and some glue around generating selectors and so for the pallet. But we should expand it at least once ^^